packages/api/src/task/scheduler.ts

```diff
@@ -378,23 +379,27 @@ export class TaskScheduler {
   async createTask(
     type: Task["type"],
     params: Task["params"],
+    projectId: string,
```

Comment:
Similarly, we should never have a `projectId` but not have a `userId`. Can you make this interface clearer?